app/main/views/broadcast.py

```diff
@@ -836,6 +841,73 @@ def reject_broadcast_message(service_id, broadcast_message_id):
         service_id=current_service.id,
     )
 
+    if broadcast_message.status != "pending-approval":
+        return redirect(
+            url_for(
+                ".view_current_broadcast",
+                service_id=current_service.id,
+                broadcast_message_id=broadcast_message.id,
+            )
+        )
+
+    form = RejectionReasonForm()
+    rejection_reason = form.rejection_reason.data
+
+    if not form.validate_on_submit():
```

Comment:
Could the logic here be swapped so we have
`if form.validate_on_submit():` followed by the try-except, then the `return render_template()`  can be the last logical line in the method.

This would keep the form handling in line with the rest of the code - it's not a big deal, but I had to pause for a second when I saw the `if not form.validate...` clause.